### PR TITLE
Connects lightWallet with redux state 

### PR DESCRIPTION
**Pull request by Moejoe90**

### CODE REVIEW

File: app/modules/storage/storage.ts [hunk 1]
```diff
@@ -14,3 +14,5 @@ export class Storage {
     return this.localStorage.getItem(key);
   };
 }
+
+//TODO: Bring back interface for storage
```
Comment (krzkaczor) [inline app/modules/storage/storage.ts]: could you just bring it back? :D it's literally 6 lines of code :D

File: app/modules/web3/LightWallet.ts [hunk 2]
```diff
@@ -89,21 +91,20 @@ export class LightWalletConnector {
   private lightWalletWeb3?: any;
   constructor(
     public readonly lightWalletVault: IVault,
-    public readonly walletSubType: WalletSubType,
     public readonly web3Config: IEthereumNetworkConfig,
   ) {}
+  public readonly walletSubType: WalletSubType = WalletSubType.UNKNOWN;
+  public readonly walletAddress: EthereumAddress = "0x".concat(
```
Comment (krzkaczor) [inline app/modules/web3/LightWallet.ts]: could you export this to separate function? like getDefaultAddressForLightWallet or whatever

File: app/modules/storage/storage.ts (hunk 1 above)
Comment (Moejoe90) [reply app/modules/storage/storage.ts]: Coming with tests :D

File: typings/modules.d.ts [hunk 3]
```diff
@@ -8,3 +8,4 @@ declare module "web3-provider-engine/subproviders/hooked-wallet";
 
 declare module "ledger-wallet-provider";
 declare module "eth-lightwallet";
+declare module "ethereumjs-util";
```
Comment (banciur) [inline typings/modules.d.ts]: There are typescipt typings for ethereumjs-util - https://github.com/machinomy/types-ethereumjs-util
not sure if they are correct but maybe worth a try.

File: app/modules/web3/LightWallet.ts (hunk 2 above)
Comment (Moejoe90) [reply app/modules/web3/LightWallet.ts]: Good note

File: typings/modules.d.ts (hunk 3 above)
Comment (Moejoe90) [reply typings/modules.d.ts]: Good that you mentioned , i tried using `@types/ethereumjs-ut`i but it didn't hold i will investigate this

File: app/modules/storage/storage.ts (hunk 1 above)
Comment (krzkaczor) [reply app/modules/storage/storage.ts]: how is it? xD

Comment (Moejoe90) [reply app/modules/storage/storage.ts]: This should be taken off interface already added

File: app/modules/wallet-selector/light-wizard/actions.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+        localStorage.setKey(LOCAL_STORAGE_LIGHT_WALLET_KEY, lightWalletVault.walletInstance);
+
+        await vaultApi.store(password, lightWalletVault.salt, lightWalletVault.walletInstance);
+        const lightWallet = await LightWalletConnector.connect(
+          {
+            walletInstance,
+            salt: lightWalletVault.salt,
+          },
+          LightWalletUtil.addPrefixToAddress(walletInstance.addresses[0]) as EthereumAddress,
```
Comment (krzkaczor) [inline app/modules/wallet-selector/light-wizard/actions.ts]: please get ethereum address by using web3 adapter inside connect

File: app/modules/wallet-selector/light-wizard/reducer.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { ILightWallet } from "../../../modules/web3/LightWallet";
+import { AppReducer } from "../../../store";
+
+export interface ILightWalletWizardState {
+  errorMsg?: string;
+  isWalletCreated: boolean;
+  isActive: boolean;
+  isSeedBacked: boolean;
+  lightWalletInstance?: ILightWallet | undefined;
```
Comment (krzkaczor) [inline app/modules/wallet-selector/light-wizard/reducer.ts]: you dont need `| undefined` here

File: app/modules/wallet-selector/light-wizard/reducer.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  salt?: string | undefined;
```
Comment (krzkaczor) [inline app/modules/wallet-selector/light-wizard/reducer.ts]: same

File: app/modules/web3/LightWallet.ts [hunk 7]
```diff
@@ -40,10 +40,75 @@ export interface ILightWallet {
...
+      const keyFromPassword = promisify<ILightWallet, string>(
+        lightWalletInstance.keyFromPassword.bind(lightWalletInstance),
+      );
+      return await keyFromPassword(password);
+    } catch (e) {
+      throw new LightWrongPasswordSaltError();
+    }
+  }
+
+  public addPrefixToAddress(address: string): string {
```
Comment (krzkaczor) [inline app/modules/web3/LightWallet.ts]: I think this won't be used anymore after you used web3adapter to obtain ethereum

File: app/modules/web3/LightWallet.ts [hunk 8]
```diff
@@ -86,102 +155,58 @@ export const LightWalletConnectorSymbol = "LightWalletConnector";
 @injectable()
 export class LightWalletConnector {
   private Web3Adapter?: any;
```
Comment (krzkaczor) [inline app/modules/web3/LightWallet.ts]: this  should be web3Adapter and replace any with Web3Adapter please.

File: test/modules/wallet-selector/light-wallet/actions.spec.ts [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+        dummyLogger,
+      );
+
+      expect(dispatchMock).to.be.calledWithExactly(walletConnectedAction);
+      expect(lightWalletConnector.connect).to.be.calledWith({
+        walletInstance: expectedWalletDummy,
+        salt: "salt",
+      });
+    });
+    it("should dispatch error action on error", async () => {
```
Comment (krzkaczor) [inline test/modules/wallet-selector/light-wallet/actions.spec.ts]: new line here please :)

File: app/components/walletSelector/walletLight/CreateWallet.tsx [hunk 10]
```diff
@@ -21,11 +21,19 @@ export const CreateWallet: React.SFC<{}> = () => {
...
-          </Link>
+          </Button>
         </Form>
       </Col>
     </Row>
   );
 };
+
+export const CreateWallet = compose<React.SFC>(
+  appConnect<any, any>({
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/walletSelector/walletLight/CreateWallet.tsx]: This is temp to run a running example if JWT obtaining from browser

File: app/modules/wallet-selector/light-wizard/actions.ts [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+        );
+        await web3Manager.plugPersonalWallet(lightWallet);
+        dispatch(obtainJwt);
+        await usersApi.createLightwalletAccount(email, lightWalletVault.salt);
+        //TODO: add error checking in case of failure
+        // redirect user to dashboard
+        dispatch(walletConnectedAction);
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/modules/wallet-selector/light-wizard/actions.ts]: What's the best idea then which action should i choose to navigate to dashboard?

File: test/modules/wallet-selector/light-wallet/actions.spec.ts (hunk 9 above)
Comment (Moejoe90) [reply test/modules/wallet-selector/light-wallet/actions.spec.ts]: Done

File: app/modules/wallet-selector/light-wizard/actions.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (krzkaczor) [reply app/modules/wallet-selector/light-wizard/actions.ts]: dont use anything now. We will solve it later a little bit later

File: app/modules/web3/LightWallet.ts (hunk 8 above)
Comment (Moejoe90) [reply app/modules/web3/LightWallet.ts]: Done

File: app/modules/web3/LightWallet.ts (hunk 7 above)
Comment (Moejoe90) [reply app/modules/web3/LightWallet.ts]: correct :)

File: app/modules/wallet-selector/light-wizard/reducer.ts (hunk 5 above)
Comment (Moejoe90) [reply app/modules/wallet-selector/light-wizard/reducer.ts]: done

File: app/modules/wallet-selector/light-wizard/reducer.ts (hunk 6 above)
Comment (Moejoe90) [reply app/modules/wallet-selector/light-wizard/reducer.ts]: done

File: app/modules/wallet-selector/light-wizard/actions.ts (hunk 4 above)
Comment (Moejoe90) [reply app/modules/wallet-selector/light-wizard/actions.ts]: taken off

File: app/modules/wallet-selector/light-wizard/actions.ts [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+export const LightWalletCreatedAction = makeActionCreator<ILightWalletCreatedAction>(
+  "LIGHT_WALLET_CREATED",
+);
+
+export const tryConnectingWithLightWallet = (email: string, password: string) =>
+  injectableFn(
+    async (
+      dispatch: AppDispatch,
+      web3Manager: Web3Manager,
+      LightWalletConnector: LightWalletConnector,
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/modules/wallet-selector/light-wizard/actions.ts]: done

File: app/modules/wallet-selector/light-wizard/actions.ts [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+      LightWalletUtil: LightWalletUtil,
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/modules/wallet-selector/light-wizard/actions.ts]: done

File: app/modules/wallet-selector/light-wizard/actions.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/modules/wallet-selector/light-wizard/actions.ts]: done

File: app/modules/wallet-selector/light-wizard/actions.ts [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+        await vaultApi.store(password, lightWalletVault.salt, lightWalletVault.walletInstance);
+        const lightWallet = await LightWalletConnector.connect(
+          {
+            walletInstance,
+            salt: lightWalletVault.salt,
+          },
+          LightWalletUtil.addPrefixToAddress(walletInstance.addresses[0]) as EthereumAddress,
+        );
+        await web3Manager.plugPersonalWallet(lightWallet);
+        dispatch(obtainJwt);
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/modules/wallet-selector/light-wizard/actions.ts]: done

**Pull request merged**